webapp/src/containers/SwapPage/index.tsx

```diff
@@ -239,7 +239,7 @@ const SwapPage: React.FunctionComponent<SwapPageProps> = (
       if (formState.hash1 === '0') {
         if (
           new BigNumber(e.target.value).lte(
-            Math.max(Number(formState.balance1) - 1, 0)
+            Math.max(new BigNumber(formState.balance1).toNumber() - 1, 0)
```

Comment:
places like this it defeats the purposes if you convert to BigNumber and then back to number